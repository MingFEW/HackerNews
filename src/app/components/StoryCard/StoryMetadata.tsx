import React from 'react'
import { BsFillStarFill } from 'react-icons/bs'
import { useTranslation } from 'react-i18next'
import 'twin.macro'

// Utils
import { formatTime } from '@/utils/formatTime'
import { messages } from './messages'

interface StoryMetadataProps {
  author: string
  time: number
  score: number
}

export const StoryMetadata: React.FC<StoryMetadataProps> = props => {
  const { t } = useTranslation()
  const { author, time, score } = props
  return (
    <div tw="flex items-center justify-between">
      <div tw="text-xs text-textGrey">{t(messages.postedBy(), { author, time: formatTime(time as number) })}</div>
      <div tw="flex items-center">
        <BsFillStarFill color="gold" tw="mr-[6px]" />
        <div tw="text-xs sm:text-sm text-white font-semibold mt-[2px]">{score}</div>
      </div>
    </div>
  )
}
