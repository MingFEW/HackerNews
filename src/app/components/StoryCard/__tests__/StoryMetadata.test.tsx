import { createRenderer } from 'react-test-renderer/shallow'

import { StoryMetadata } from '../StoryMetadata'

const shallowRenderer = createRenderer()

describe('<StoryMetadata />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<StoryMetadata author="test" time={1673919481} score={10} />)
    const renderedOutput = shallowRenderer.getRenderOutput()
    expect(renderedOutput).toMatchSnapshot()
  })
})
