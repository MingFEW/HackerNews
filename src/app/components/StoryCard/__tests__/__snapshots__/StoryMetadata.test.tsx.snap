// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoryMetadata /> should render and match the snapshot 1`] = `
<div
  tw="flex items-center justify-between"
>
  <div
    tw="text-xs text-textGrey"
  >
    _NOT_TRANSLATED_
  </div>
  <div
    tw="flex items-center"
  >
    <BsFillStarFill
      color="gold"
      tw="mr-[6px]"
    />
    <div
      tw="text-xs sm:text-sm text-white font-semibold mt-[2px]"
    >
      10
    </div>
  </div>
</div>
`;
